lte/gateway/c/sctpd/sctpd_downlink_impl.cpp

```diff
@@ -35,34 +36,48 @@ Status SctpdDownlinkImpl::Init(
   InitRes *res)
 {
   MLOG(MDEBUG) << "SctpdDownlinkImpl::Init starting";
-
-  if (_sctp_connection != nullptr && !req->force_restart()) {
-    MLOG(MINFO) << "SctpdDownlinkImpl::Init reusing existing connection";
-    res->set_result(InitRes::INIT_OK);
-    return Status::OK;
-  }
-
-  if (_sctp_connection != nullptr) {
-    MLOG(MDEBUG)
-      << "SctpdDownlinkImpl::Init cleaning up sctp_desc and listener";
-
-    auto conn = std::move(_sctp_connection);
-    conn->Close();
-  }
-
-  MLOG(MDEBUG) << "SctpdDownlinkImpl::Init creating new socket and listener";
-
-  try {
-    _sctp_connection = std::make_unique<SctpConnection>(*req, _uplink_handler);
-  } catch (...) {
-    res->set_result(InitRes::INIT_FAIL);
-    return Status::OK;
-  }
-
-  _sctp_connection->Start();
-
-  res->set_result(InitRes::INIT_OK);
-  return Status::OK;
+  if (req->ppid() == S1AP) {
+	   if (_sctp_connection != nullptr && !req->force_restart()) {
+		    MLOG(MINFO) << "SctpdDownlinkImpl::Init reusing existing connection";
+		    res->set_result(InitRes::INIT_OK);
+		    return Status::OK;
+	   }
+	   if (_sctp_4G_connection != nullptr) {
+		    MLOG(MDEBUG)<< "SctpdDownlinkImpl::Init cleaning up sctp_desc and listener";
+		    auto conn = std::move(_sctp_connection);
+		    conn->Close();
+	   }
+	   MLOG(MDEBUG) << "SctpdDownlinkImpl::Init creating new socket and listener";
+	   try {
+		    _sctp_4G_connection = std::make_unique<SctpConnection>(*req, _uplink_handler);
+		     } catch (...) {
+			      res->set_result(InitRes::INIT_FAIL);
+			      return Status::OK;
+		     }
+	    _sctp_4G_connection->Start();
+  }else if (req->ppid() == NGAP) {
+	  MLOG(MDEBUG) << "SctpdDownlinkImpl::Init starting for 5G";
+	  if (_sctp_5G_connection != nullptr && !req->force_restart()) {
+		  MLOG(MINFO) << "SctpdDownlinkImpl::Init reusing existing 5G connection";
+		  res->set_result(InitRes::INIT_OK);
+		  return Status::OK;
+	  }
+	  if (_sctp_5G_connection != nullptr) {
+		  MLOG(MDEBUG<< "SctpdDownlinkImpl::Init cleaning up sctp_desc and listener of 5G";
+	          auto conn = std::move(_sctp_5G_connection);
+		  conn->Close();
+		  }
+		  MLOG(MDEBUG) << "SctpdDownlinkImpl::Init creating new socket and listener of 5G";
+		  try {
+		  	_sctp_5G_connection = std::make_unique<SctpConnection>(*req, _uplink_handler);
+			} catch (...) {
+			res->set_result(InitRes::INIT_FAIL);
+			return Status::OK;
+			}
+			_sctp_5G_connection->Start();
+	}
+	res->set_result(InitRes::INIT_OK);
+	return Status::OK;
```

Comment:
did you run clang formatter on this?

Reply:
ulas kindly refer [5G: MVC][Sctpd][Sctp] 5g Association
#2669 for the changes